components/devtools/actors/network_event.rs

```diff
@@ -79,29 +167,124 @@ impl Actor for NetworkEventActor {
         Ok(match msg_type {
             "getRequestHeaders" => {
                 // TODO: Pass the correct values for headers, headerSize, rawHeaders
+                let headersSize = self.request.headers.len() as u8;
+                let mut headerNames = Vec::new();
+                let mut rawHeadersString = "".to_owned();
+                for item in self.request.headers.iter() {
+                    let name = item.name();
+                    let value = item.value_string();
+                    headerNames.push(name.to_owned());
+                    rawHeadersString = rawHeadersString + name + ":" + &value + "\r\n";
+                }
                 let msg = GetRequestHeadersReply {
                     from: self.name(),
-                    headers: Vec::new(),
-                    headerSize: 10,
-                    rawHeaders: "Raw headers".to_owned(),
+                    headers: headerNames,
+                    headerSize: headersSize,
+                    rawHeaders: rawHeadersString,
                 };
                 stream.write_json_packet(&msg);
                 ActorMessageStatus::Processed
             }
             "getRequestCookies" => {
-                ActorMessageStatus::Ignored
+                let mut cookies = Vec::new();
+                    if let Some(req_cookies) = self.request.headers.get_raw("Cookie") {
+                        for cookie in &*req_cookies {
+                            if let Ok(cookie_value) = String::from_utf8(cookie.clone()) {
+                                cookies = cookie_value.into_bytes();
+                            }
+                        }
+                }
```

Comment:
It's actually the preceding block (starting with the `if let`) that is over-indented by four spaces.